src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -697,6 +705,15 @@ private static BuildEventArgsFieldFlags GetBuildEventArgsFieldFlags(BuildEventAr
             if (writeMessage)
             {
                 flags |= BuildEventArgsFieldFlags.Message;
+
+                // We're only going to write the arguments for messages,
+                // warnings and errors. Only set the flag for these.
+                if (e is LazyFormattedBuildEventArgs lazyFormattedBuildEventArgs &&
+                    lazyFormattedBuildEventArgs.RawArguments is { Length: > 0 } &&
+                    (e is BuildMessageEventArgs or BuildWarningEventArgs or BuildErrorEventArgs))
```

Comment:
You can take this even further if you want, doing away with the temporary variable altogether and using a single pattern:

```suggestion
                if (e is LazyFormattedBuildEventArgs { RawArguments: { Length: > 0 } } and
                    (BuildMessageEventArgs or BuildWarningEventArgs or BuildErrorEventArgs))
```